Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/features/input/model/DolphinVibratorManagerCompat.kt

```diff
@@ -0,0 +1,24 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+package org.dolphinemu.dolphinemu.features.input.model
+
+import android.os.Vibrator
+
+class DolphinVibratorManagerCompat(vibrator: Vibrator?) : DolphinVibratorManager {
+    private val vibrator: Vibrator
+    private val vibratorIds: IntArray
+
+    init {
+        this.vibrator = vibrator!!
```

Comment:
Ah ok, I got confused since immediately after this you use the `vibrator` reference to check if a vibrator is present. In that case, would it be alright to use `vibrator?.hasVibrator() == true` to prevent a `NullPointerException`? And then should I add `!!` to the vibrator returned in `getVibrator()` or should it be nullable?

Reply:
Sorry, I must have been misreading the Android documentation when writing my review. The vibrator can't be null. I would suggest changing the class parameter type from `Vibrator?` to `Vibrator`.